core/camel-support/src/main/java/org/apache/camel/support/cache/DefaultProducerCache.java

```diff
@@ -173,7 +177,12 @@ public void releaseProducer(Endpoint endpoint, AsyncProducer producer) {
     @Override
     public Exchange send(Endpoint endpoint, Exchange exchange, Processor resultProcessor) {
         if (camelContext.isStopped()) {
-            exchange.setException(new RejectedExecutionException("CamelContext is stopped"));
+            if (!filledStackTrace) {
+                REJECTED_EXECUTION_EXCEPTION.fillInStackTrace();
+                filledStackTrace = true;
```

Comment:
-1 to such strange code to happen during stopping.

We should only optimize for core routing engine (keep basic code clean and simple).

Reply:
Agreed. Let's rollback on this one as it's too much effort for something minor.